scripts/metadata_comparison/extractor.py

```diff
@@ -0,0 +1,193 @@
+    # destination_blob_name = "storage/object/name"
+
+    bucket = gcs_storage_client.bucket(bucket_name)
+
+    print('Uploading file content to gs://{bucket}/{dest}...'.format(bucket=bucket_name, dest=destination_blob_name), end='')
+    blob = bucket.blob(destination_blob_name)
+    blob.upload_from_string(source_file_contents)
+    print(" done!")
+
+
+def upload_workflow_metadata_json(bucket_name, raw_workflow_metadata, workflow_gcs_base_path, gcs_storage_client):
+    workflow_gcs_metadata_upload_path = '{workflow_gcs_base_path}/metadata.json'.format(workflow_gcs_base_path=workflow_gcs_base_path)
+    upload_blob(bucket_name, raw_workflow_metadata, workflow_gcs_metadata_upload_path, gcs_storage_client)
+
+
+def upload_operations_metadata_json(bucket_name, operation_id, operations_metadata, workflow_gcs_base_path, gcs_storage_client):
+    """Uploads metadata to cloud storage, as json"""
+    operation_upload_path = '{workflow_gcs_base_path}/operations/{miniid}.json'.format(workflow_gcs_base_path=workflow_gcs_base_path, miniid=get_operation_id_number(operation_id))
+    formatted_metadata = json.dumps(operations_metadata, indent=2)
+    upload_blob(bucket_name, formatted_metadata, operation_upload_path, gcs_storage_client)
+
+
+def process_workflow(cromwell_url, gcs_bucket, gcs_path, gcs_storage_client, genomics_client, workflow):
+    raw_metadata, json_metadata = fetch_raw_workflow_metadata(cromwell_url, workflow)
+    workflow_gcs_base_path = '{root_gcs_path}/{wfid}/extractor'.format(root_gcs_path=gcs_path, wfid=workflow)
+
+    operation_ids = find_operation_ids_in_metadata(json_metadata)
+    operation_metadata = { id: read_operations_metadata(id, api, genomics_client) for (id, api) in operation_ids.items() }
+    for id in operation_metadata:
+        upload_operations_metadata_json(gcs_bucket, id, operation_metadata[id], workflow_gcs_base_path, gcs_storage_client)
+    upload_workflow_metadata_json(gcs_bucket, raw_metadata, workflow_gcs_base_path, gcs_storage_client)
+
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(
+        description='Extract metadata and operation details for workflows and upload to GCS')
+    parser.add_argument('cromwell_url', metavar='CROMWELL', type=url_regex_validator, nargs=1, help='Cromwell host')
+    parser.add_argument('gcs_path', metavar='GCSPATH', type=gcs_path_regex_validator, nargs=1, help='GCS path to upload to')
+    parser.add_argument('workflows', metavar='WORKFLOW', type=workflow_regex_validator, nargs='+', help='Workflows to process')
+
+    args = parser.parse_args()
+    cromwell_url = args.cromwell_url[0]
+    gcs_bucket, gcs_path = args.gcs_path[0]
+    workflows = args.workflows
+
+    credentials, project_id = google.auth.default()
+    storage_client = storage.Client(credentials = credentials)
+    genomics_v1_client = build('genomics', 'v1alpha2', credentials = credentials)
+    genomics_client = build('genomics', 'v2alpha1', credentials = credentials)
```

Comment:
it looks like we don't really need to have two genomics clients which is fortunate given the sketchiness of the provenance...